### PR TITLE
Update PackageGet.php

### PR DESCRIPTION
**Pull request by felixk3y**
fixed bug

### CODE REVIEW

File: Sources/PackageGet.php [hunk 1]
```diff
@@ -652,12 +652,14 @@ function PackageUpload()
 
 	// Make sure it has a sane filename.
 	$_FILES['package']['name'] = preg_replace(array('/\s/', '/\.[\.]+/', '/[^\w_\.\-]/'), array('_', '.', ''), $_FILES['package']['name']);
-
-	if (strtolower(substr($_FILES['package']['name'], -4)) != '.zip' && strtolower(substr($_FILES['package']['name'], -4)) != '.tgz' && strtolower(substr($_FILES['package']['name'], -7)) != '.tar.gz')
+	$extension = substr(strrchr(strtolower($_FILES['package']['name']), '.'), 0);
+	if(!in_array($extension, $allowext)){
```
Comment (live627) [inline Sources/PackageGet.php]: curly brace should be on its own line

**Comment by live627** [comment]
commit must be signed off

**Comment by MissAllSunday** [comment]
@felixk3y Please sign up your commit otherwise we can't merge your PR.

File: Sources/PackageGet.php [hunk 2]
```diff
@@ -652,12 +652,15 @@ function PackageUpload()
...
+	$extension = substr(strrchr(strtolower($_FILES['package']['name']), '.'), 0);
+	if(!in_array($extension, $allowext))
+	{
 		fatal_lang_error('package_upload_error_supports', false, array('zip, tgz, tar.gz'));
-
+	}
+	
 	// We only need the filename...
-	$packageName = basename($_FILES['package']['name']);
+	$extension = ($extension == '.gz') ? '.tar.gz' : $extension ;
```
Comment (matthew-kerle) [inline Sources/PackageGet.php:662]: Should be ...`$extension;` instead of `$extension ;`

File: Sources/PackageGet.php [hunk 3]
```diff
@@ -652,12 +652,15 @@ function PackageUpload()
 
 	// Make sure it has a sane filename.
 	$_FILES['package']['name'] = preg_replace(array('/\s/', '/\.[\.]+/', '/[^\w_\.\-]/'), array('_', '.', ''), $_FILES['package']['name']);
-
-	if (strtolower(substr($_FILES['package']['name'], -4)) != '.zip' && strtolower(substr($_FILES['package']['name'], -4)) != '.tgz' && strtolower(substr($_FILES['package']['name'], -7)) != '.tar.gz')
+	$extension = substr(strrchr(strtolower($_FILES['package']['name']), '.'), 0);
+	if(!in_array($extension, $allowext))
```
Comment (matthew-kerle) [inline Sources/PackageGet.php:656]: Should be `if (` not `if(`

File: Sources/PackageGet.php [hunk 4]
```diff
@@ -641,7 +641,7 @@ function PackageUpload()
 
 	// Setup the correct template, even though I'll admit we ain't downloading ;)
 	$context['sub_template'] = 'downloaded';
-
+	$allowext = array('.zip','.tgz','.gz');
```
Comment (matthew-kerle) [inline Sources/PackageGet.php:644]: SMF prefers spaces after commas in arrays on one line between elements. `$allowext = array('.zip', '.tgz', '.gz');`

**Comment by colinschoen** [comment]
Are you planning on making these changes, feilxk3y?

**Comment by XinYenFon** [comment]
@live627 if anything here worth cherry picking lets do it, or close this.

**Pull request merged**